processor/src/main/java/org/mapstruct/ap/spi/DefaultAccessorNamingStrategy.java

```diff
@@ -102,10 +102,24 @@ public boolean isSetterMethod(ExecutableElement method) {
     protected boolean isBuilderSetter(ExecutableElement method) {
         return method.getParameters().size() == 1 &&
             !JAVA_JAVAX_PACKAGE.matcher( method.getEnclosingElement().asType().toString() ).matches() &&
+            !isAdderWithUpperCase4thCharacter( method ) &&
             //TODO The Types need to be compared with Types#isSameType(TypeMirror, TypeMirror)
             method.getReturnType().toString().equals( method.getEnclosingElement().asType().toString() );
     }
 
+    /**
+     * Checks that the method is an adder with an upper case 4th character. The reason for this is that methods such
+     * as {@code address(String address)} are considered as setter and {@code addName(String name)} too. We need to
```

Comment:
ahh... tricky..

Reply:
Can't we derive the fact that its an adder by means of this Singular / Plural thing and by means of the fact that there 's also a getter returning a list of what the adder adds? Feels a bit more safe than relying on camelcasing..

Reply:
That is in the internal packages and this is in the public package. I don't think that it is a good idea to expose that here.

Reply:
True.. We should not expose it directly.. But we could expose perhaps expose it as util (hiding the implementation).. WDYT?

Reply:
I disagree. The accessor naming strategy decides based only on a method whether it can be a getter, setter, adder or something else. It doesn't know whether a getter (setter) for the given adder exists.